src/metabase/search/impl.clj

```diff
@@ -210,13 +203,15 @@
 (defn default-engine
   "In the absence of an explicit engine argument in a request, which engine should be used?"
   []
-  (if config/is-test?
-    ;; TODO The API tests have not yet been ported to reflect the new search's results.
-    :search.engine/in-place
-    (if-let [s (public-settings/search-engine)]
-      (u/prog1 (keyword "search.engine" (name s))
-        (assert (search.engine/supported-engine? <>)))
-      :search.engine/in-place)))
+  (if-let [s (public-settings/search-engine)]
+    (let [engine (keyword "search.engine" (name s))]
+      (if (search.engine/supported-engine? engine)
+        engine
+        ;; It would be good to have a warning on start up for this.
```

Comment:
I think this needs to be another post handover TODO unfortunately, as I don't think its trivial.